### PR TITLE
Do not show marketplace icon for joining creature dialog is none is built

### LINKED ISSUE #2654: Adventure map, joining creature window with no marketplace built
If we don't have any marketplace we can still press the place, where the button is supposed to be and enter exchange window.
But we still cannot trade resources.
image
image
image

### PR DESCRIPTION
**Pull request by vincent-grosbois**
close #2654

### CODE REVIEW

File: src/fheroes2/dialog/dialog_armyinfo.cpp [hunk 1]
```diff
@@ -706,18 +706,18 @@ int Dialog::ArmyJoinWithCost( const Troop & troop, u32 join, u32 gold, Heroes &
...
-            std::string msg = _( "Not enough gold (%{gold})" );
-            StringReplace( msg, "%{gold}", gold - kingdom.GetFunds().Get( Resource::GOLD ) );
-            tsNotEnoughGold.SetText( msg, Font::SMALL );
-            tsNotEnoughGold.SetPos( btnMarketArea.x - 25, btnMarketArea.y - 17 );
-            tsNotEnoughGold.Show();
-            btnMarket.draw();
-        }
+    if ( kingdom.AllowPayment( payment_t( Resource::GOLD, gold ) ) || kingdom.GetCountMarketplace() == 0 ) {
+        tsNotEnoughGold.Hide();
+        btnMarket.disable();
```
Comment (ihhub) [inline src/fheroes2/dialog/dialog_armyinfo.cpp:713]: It'll be more safe to call `btnMarket.hide();` as well then we won't draw this button.

Comment (ihhub) [inline src/fheroes2/dialog/dialog_armyinfo.cpp:713]: It'll be more safe to call btnMarket.hide(); as well then we won't draw this button.

File: src/fheroes2/dialog/dialog_armyinfo.cpp [hunk 2]
```diff
@@ -778,14 +778,17 @@ int Dialog::ArmyJoinWithCost( const Troop & troop, u32 join, u32 gold, Heroes &
...
             tsNotEnoughGold.Hide();
+            btnMarket.disable();
         }
         else {
             std::string msg = _( "Not enough gold (%{gold})" );
             StringReplace( msg, "%{gold}", gold - kingdom.GetFunds().Get( Resource::GOLD ) );
             tsNotEnoughGold.SetText( msg, Font::SMALL );
             tsNotEnoughGold.Show();
+            btnMarket.enable();
+            btnMarket.draw();
```
Comment (ihhub) [inline src/fheroes2/dialog/dialog_armyinfo.cpp:791]: Please move `btnMarket.draw();` outside this if-body.

File: src/fheroes2/dialog/dialog_armyinfo.cpp [hunk 3]
```diff
@@ -778,14 +778,17 @@ int Dialog::ArmyJoinWithCost( const Troop & troop, u32 join, u32 gold, Heroes &
 
         btnGroup.draw();
 
-        if ( allowPayment ) {
+        if ( allowPayment || kingdom.GetCountMarketplace() == 0 ) {
```
Comment (ihhub) [inline src/fheroes2/dialog/dialog_armyinfo.cpp:781]: If we hide the button before the loop then we don't need `kingdom.GetCountMarketplace() == 0` condition.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @vincent-grosbois , I left few comments here. Would you mind to take a look?

**Comment by Branikolog** [comment]
Hello, @vincent-grosbois 
A "castle" button with marketplace still appears, if we don't have market built in town and some creatures are willing to join us.
image

**Comment by ihhub** [comment]
@vincent-grosbois , thank you very much for the fix!

**Pull request merged**